src/DetailsView/reports/components/report-file-name-generator.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+type DateProvider = () => Date;
+
+export type ReportFileNameGeneratorDeps = {
+    dateProvider: DateProvider;
+};
+
+export class ReportFileNameGenerator {
+    constructor(private readonly deps: ReportFileNameGeneratorDeps) {}
+
+    public getFileName(baseName: string, extension: string): string {
+        const { dateProvider } = this.deps;
+        const isoString = dateProvider().toISOString();
+        const dateTime = `${isoString.substr(0, 10)} ${isoString.substr(11, 8)}`;
```

Comment:
I noticed that this generates a GMT date/time for the file name, whereas the date/time shown in the report is local time. Should the two match?

Reply:
Another reason the timestamps might diverge for automated checks between the report and the file name: the report shows the time of the scan, and the file name is the time of the export. (Assessment reports currently display the time of report generation, so they'd only be different if a displayed time boundary is crossed.)

Reply:
Back to local time.